examples/petsc/bp1.cu

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+// All Rights reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+// *****************************************************************************
+extern "C" __global__ void Setup(void *ctx, CeedInt Q,
+                                 Fields_Cuda fields) {
+  CeedScalar *rho = fields.outputs[0], *true_soln = fields.outputs[1], *rhs = fields.outputs[2];
```

Comment:
I am ok with merging this third strategy for managing inputs/outputs to QFunction for now, but I think we are long overdue for getting down to one QFunction source. libCEED simply won't be usable if 3 copies of every QFunction have to be maintianed and diverging strategies will impeed the full QFunction Gallery.

I think single source needs to be our next objective after merging the new backends.

@jedbrown, thoughts on adding a third input/output strategy for QFunctions?